packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -942,6 +947,10 @@ export class PromoteService extends BaseService {
 
             for await (const spaceChange of filteredSortedSpaceChanges) {
                 if (newSpaces.has(spaceChange.data.path)) {
+                    parentSpaceUuid = newSpaces.get(
+                        spaceChange.data.path,
+                    )!.uuid;
```

Comment:
instead of ! should we check and throw instead?

Reply:
we check that it exists right above so should be good